tests/bsim/bluetooth/mesh/src/mesh_test.h

```diff
@@ -122,6 +122,23 @@
 		}                                                                                  \
 	} while (0)
 
+#define WAIT_FOR_COND(cond, wait)                                                                  \
```

Comment:
Seems like an attempt to create your own thread synchronization approach.
Is it possible to use conditional variable or event API:
https://docs.zephyrproject.org/apidoc/latest/group__condvar__apis.html
https://docs.zephyrproject.org/apidoc/latest/group__event__apis.html

Reply:
Yes, I just needed something very simple to sporadically check if some condition is fullfilled within a certain time. It could have been made a lot better, but I think it works fine for many non-time critical checks like the ones in this test.